### PR TITLE
Update fetch and generate scripts for reflection

### PR DESCRIPTION
**Pull request by glbrntt**
Motivation:

To add the reflection service we need to fetch it and generate the appropriate code.

Modifications:

- Update proto fetching script to get v1 reflection service
- Update proto generating script to generate reflection service code and messages
- Also add protos used in tests for the reflection service and generate their descriptor sets

Result:

Generated code is in place

### CODE REVIEW

File: dev/protos/generate.sh [hunk 1]
```diff
@@ -94,10 +94,65 @@ function generate_health_service {
   generate_grpc "$proto" "$(dirname "$proto")" "$output" "Visibility=Package" "Client=true" "Server=true" "UseAccessLevelOnImports=true"
 }
 
+function generate_reflection_service {
+  local proto="$here/upstream/grpc/reflection/v1/reflection.proto"
+  local output="$root/Sources/GRPCReflectionService/Generated"
+
+  # Messages were accidentally leaked into public API, they shouldn't be but we
+  # can't undo that change until the next major version.
```
Comment (gjcairo) [inline dev/protos/generate.sh:102]: Does this still apply?

File: dev/protos/generate.sh [hunk 2]
```diff
@@ -94,10 +94,65 @@ function generate_health_service {
...
+  local proto="$here/upstream/grpc/health/v1/health.proto"
+  local proto_path="$here/upstream"
+  local output="$root/Tests/GRPCReflectionServiceTests/Generated/DescriptorSets/health.pb"
+
+  invoke_protoc --descriptor_set_out="$output" "$proto" -I "$proto_path" \
+    --include_source_info \
+    --include_imports
+}
+
+function generate_extended_message_descriptor_set {
```
Comment (gjcairo) [inline dev/protos/generate.sh:129]: Should this be `base_message`?
```suggestion
function generate_base_message_descriptor_set {
```

File: dev/protos/generate.sh (hunk 1 above)
Comment (glbrntt) [reply dev/protos/generate.sh:102]: Good catch, this was copy-pasta!

Comment (glbrntt) [inline dev/protos/generate.sh:102]:
```suggestion
```

File: dev/protos/generate.sh [hunk 3]
```diff
@@ -94,10 +94,65 @@ function generate_health_service {
...
-# Interoperability tests
 generate_interop_test_service
-
-# Health service
 generate_health_service
+generate_reflection_service
+
+generate_reflection_service_descriptor_set
+generate_health_service_descriptor_set
+generate_extended_message_descriptor_set
```
Comment (glbrntt) [inline dev/protos/generate.sh:157]:
```suggestion
generate_base_message_descriptor_set
```

**Review by gjcairo — APPROVED** [review]

**Pull request merged**